### DISCUSSION TITLE
RSSI from a Sleepy End Device (SED)

### DISCUSSION THREAD

**Discussion by ksathara** [openthread/openthread#8946] [Q&A]
Hi,

I want to get average RSSI from the messages of a SED. I am trying to use otMessageGetRss (), to receive this from the last message. In my case the SED will be sending messages to a router/leader and I can get this info from them. But the issue is if my SED is connecting through a router, then I have to get this from the rouer -> leader. So I am thinking if it is possible for the SED to send this info as part of the payload. Is it possible to get the averaged RSSI from any API in the SED?

Thanks,
Kaushalya

**Comment by Irving-cl** [comment]
Hi @ksathara,

Do you mean to get RSSI of SED on another device other than the SED's parent? For example:
```
    leader <----> router <----> SED
```
And you want to see SED's RSSI on the leader?

**Comment by ksathara** [comment]
Hi Li,

Yeah I want the RSSI of the SED from the leader. I can think of getting
this info from the router to the leader, but this is cumbersome. If I can
send the RSSI from SED in its payload, then the routing does not matter.

Cheers,
Kaushalya

**Comment by Irving-cl** [comment]
@ksathara

>  If I can send the RSSI from SED in its payload, then the routing does not matter.

I assume you mean adding this information in your application level message and send it from the SED to the leader. And you want to get this information (the RSSI of SED seen on the router) from some OT API on the SED.

Note that the RSSI of the SED is seen on the router. So the SED doesn't have direct access to the information. However I think you can use the [link metrics feature](https://github.com/openthread/openthread/tree/main/src/cli#linkmetrics-mgmt-ipaddr-enhanced-ack-clear) which enables a node to know the quality of the link, including the average RSSI seen on the peer. The feature provides 3 ways to know the link metrics: 1. Single Probe 2. Forward Tracking Series 3. Enhanced ACK-based Probing You may want to refer the Thread spec 4.11 for detailed information.

For your case, you can use this feature to let the SED know the RSSI of SED seen on the router. And you can send this information in your application messages. The ot APIs to use this feature is at: [link_metrics.h](https://github.com/openthread/openthread/blob/main/include/openthread/link_metrics.h)

**Comment by ksathara** [comment]
Hi Li,

Much appreciated. Can you point me to an example project where link metrics
is used?

heers,
Kaushalya

**Reply by Irving-cl** [reply]
@ksathara,

Sorry I don't know any external example project that uses link metrics. But I think the best example should be cli app in Openthread. The cli app is an application that simply runs the Thread stack and provides command line support for OT public APIs to control the Thread instance. So you can reference the usage of link metrics feature in [cli module source code](https://github.com/openthread/openthread/blob/main/src/cli/cli.cpp#L4493). For example, how to use the OT API to make a link metrics query: [code](https://github.com/openthread/openthread/blob/main/src/cli/cli.cpp#L4619)

**Comment by jwhui** [comment]
It is also possible to collect RSSI / Link Quality / Link Margin information from the parent(s). Check out the enhancements in:
- https://github.com/openthread/openthread/pull/8866
